elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryDAO.java

```diff
@@ -48,7 +49,14 @@ public AsyncQuery updateAsyncQueryResult(AsyncQueryResult asyncQueryResult,
      * @param status status to be updated
      * @return query object list updated
      */
-    public Collection<AsyncQuery> updateStatusAsyncQueryCollection(String filterExpression,
+    public Collection<AsyncQuery> updateStatusAsyncQueryCollection(FilterExpression filterExpression,
             QueryStatus status);
+    /**
+     * This method gets a collection of AsyncQuery and objects from database with status QUEUED/PROCESSING/CANCELLED and
+     * returns the objects.
+     * @param filterExpression filter expression to cancel AsyncQuery Objects based on
+     * @return query object list updated
+     */
+    public Collection<AsyncQuery> getActiveAsyncQueryCollection(FilterExpression filterExpression);
```

Comment:
The DAO will be more usable if we don't tie down the contract with constraints like limiting the status to QUEUED, PROCESSING, etc.

This can just be a load method that takes a filter expression.  Let the calling code determine what states or other bits to filter on.